contract/models/account_analytic_contract_line.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountAnalyticContractLine(models.Model):
+
+    _name = 'account.analytic.contract.line'
+    _description = 'Contract Lines'
+    _inherit = 'account.analytic.invoice.line'
+
+    analytic_account_id = fields.Many2one(
+        string='Contract',
+        comodel_name='account.analytic.contract',
+        required=True,
+        ondelete='cascade',
+    )
+
+    @api.multi
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
```

Comment:
The `_onchange_product_id` in `account.analytic.invoice.line` uses the partner pretty heavily, which isn't available in the `contract.line`. This method performs similar operation, just with the user partner instead.

I guess alternatively I could just add some try/excepts in the invoice line. I'll try that